### ISSUE TITLE
Training not exceed GPU-RAM, but Evaluation exceed GPU-RAM?

### ISSUE THREAD

**Issue by williamhyin** [google/automl#765]
Hi all,

I use V100 to train my datasets which have 200 classes with batch_size=4 on Efficientdet D3.
I have no problem during training.
But when i want to evaluate the trained model.
It show that 

`2020-09-13 12:42:32.704730: W tensorflow/core/framework/cpu_allocator_impl.cc:81] Allocation of 5809176576 exceeds 10% of free system memory.
`
The following are traing and eval command.

```
d="d3"
times="dfg_xla"

!python efficientdet/main.py --mode=eval \
--model_name=efficientdet-{d} \
--model_dir=efficientdet/finetune{d}_{times} \
--validation_file_pattern=test_tfrecord/test* \
--ckpt=efficientdet/finetune{d}_{times} \
--val_json_file=labels/test.json \
--hparams=efficientdet/configs/dfg.yaml \
--eval_batch_size=1 \
--eval_samples=1703 \

!python efficientdet/main.py --mode=train\
    --training_file_pattern=train_tfrecord/train* \
    --model_name=efficientdet-{d} \
    --model_dir=efficientdet/finetune{d}_{times}  \
    --ckpt=efficientdet/weights/efficientdet-{d}  \
    --train_batch_size=4 \
    --num_examples_per_epoch=14029 --num_epochs=50  \
    --hparams=efficientdet/configs/dfg.yaml \
    --save_checkpoints_steps=3500 \

config:
num_classes: 201
label_map: {0: background, ......., 200: I-1}
use_keras_model: False
image_size: 896
mixed_precision: True
max_instances_per_image: 201
nvgpu_logging: True

```
```
Limit:                     15473775744
InUse:                     10734573824
MaxInUse:                  13605005056
NumAllocs:                        2477
MaxAllocSize:               5809176576
Reserved:                            0
PeakReserved:                        0
LargestFreeBlock:                    0
```

What might be the issue..? 
Thanks!

**Comment by Pari-singh** [comment]
I get the same error. And I can confirm that this is happening only with the recent commits, as I was able to successfully evaluate earlier (although the classwise AP were 0)
Could the author, @mingxingtan comment on the behavior they expected during the recent commits.

**Comment by mingxingtan** [comment]
Thanks for reporting the problem.

The issue is from the non-tpu estimator, which only support one batch size, so we ended up using train_batch_size for both train and eval. In your eval command line, because you did not specify train_batch_size, it use the default 64 as eval batch size, causing OOM.

I have fixed the issue. If you sync to the head and rerun your eval command line, it should run well.
